nixos/doc/manual/development/settings-options.xml

```diff
@@ -167,13 +167,51 @@ in {
 
     # We know that the `user` attribute exists because we set a default value
     # for it above, allowing us to use it without worries here
-    users.users.${cfg.settings.user} = {}
+    users.users.${cfg.settings.user} = {};
 
     # ...
   };
 }
 </programlisting>
    </example>
+   <section xml:id="sec-settings-attrs-options">
+    <title>Option declarations for attributes</title>
+    <para>
+     It can be very valuable to declare options for a subset of <literal>settings</literal> attributes in order to check their type and show their documentation in the manual. This can be done using <xref linkend='sec-freeform-modules'/>.
+     <example xml:id="ex-settings-typed-attrs">
+      <title>Declaring a type-checked <literal>settings</literal> attribute</title>
+      <para>
+       We extend above example using freeform modules to declare an option for the port, which will enforce it to be a valid integer and make it show up in the manual.
+      </para>
+<programlisting>
+settings = lib.mkOption {
+  # Setting this type allows for correct merging behavior
```

Comment:
```suggestion
```

Although defaults can be considered _merging behavior_ it's not so evident from the example. I've mentioned merging behavior in an earlier suggestion, so that may be sufficient to cover it. Though perhaps a different example can demonstrate the value of custom merging behavior?